diff --git a/meta-facebook/meta-backports/rocko/classes/kernel.bbclass b/meta-facebook/meta-backports/rocko/classes/kernel.bbclass
index 2c805b4003..858940e467 100644
--- a/meta-facebook/meta-backports/rocko/classes/kernel.bbclass
+++ b/meta-facebook/meta-backports/rocko/classes/kernel.bbclass
@@ -329,7 +329,7 @@ kernel_do_install() {
 	if (grep -q -i -e '^CONFIG_MODULES=y$' .config); then
 		oe_runmake DEPMOD=echo MODLIB=${D}${nonarch_base_libdir}/modules/${KERNEL_VERSION} INSTALL_FW_PATH=${D}${nonarch_base_libdir}/firmware modules_install
 		rm "${D}${nonarch_base_libdir}/modules/${KERNEL_VERSION}/build"
-		rm "${D}${nonarch_base_libdir}/modules/${KERNEL_VERSION}/source"
+		rm -f "${D}${nonarch_base_libdir}/modules/${KERNEL_VERSION}/source"
 		# If the kernel/ directory is empty remove it to prevent QA issues
 		rmdir --ignore-fail-on-non-empty "${D}${nonarch_base_libdir}/modules/${KERNEL_VERSION}/kernel"
 	else
